docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,93 @@
+# Topology-aware Traffic Routing
+
+The `TopologyAwareRouting` feature enables topology-aware traffic routing for components in the garden or seed clusters.
+
+## Motivation
+
+The enablement of [highly available Shoot control planes](shoot_high_availability.md) requires multi-zone Seed clusters. Garden cluster can also be a multi-zone cluster. The topology-aware routing is introduced to reduce costs and to improve network performance for the cross availability zone traffic. The cross availability zone traffic is charged by the cloud providers and it comes with higher latency compared to the traffic within the same zone. The `TopologyAwareRouting` feature enables topology-aware routing for Services deployed in the garden or seed clusters. For the clients consuming these topology-aware Services, kube-proxy prefers the endpoints which are within the same zone.
+
+## How it works?
+
+The `TopologyAwareRouting` feature relies on the Kubernetes feature [`TopologyAwareHints`](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/).
+
+##### EndpointSlice Hints mutating webhook
+
+The component that is responsible for providing hints in the EndpointSlices resources is the [kube-controller-manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), in particular this is the [EndpointSlice controller](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/). However, there are several drawbacks with the TopologyAwareHints feature that don't allow us to use it in its native way:
+- The algorithm in the EndpointSlice controller is based on a CPU-balance heuristic. From the TopologyAwareHints documentation:
+   > The controller allocates a proportional amount of endpoints to each zone. This proportion is based on the allocatable CPU cores for nodes running in that zone. For example, if one zone had 2 CPU cores and another zone only had 1 CPU core, the controller would allocate twice as many endpoints to the zone with 2 CPU cores.
+
+   In case it is not possible to achieve a balanced distribution of the endpoints, as a safeguard mechanism the controller removes hints from the EndpointSlice resource.
+   In our setup the clients and the servers are well-known and usually the traffic a component receives does not depend on the zone's allocatable CPU. 
+   Most of the components deployed by Gardener are scaled automatically by VPA (in most of the cases). In case of an overload of a replica, the VPA should provide and apply enhanced CPU and memory resources. Additionally, Gardener uses the cluster-autoscaler to upscale/downscale Nodes dynamically. Hence, it is not possible to ensure a balanced of allocatable CPU accross the zones.
```

Comment:
```suggestion
   Most of the components deployed by Gardener are scaled automatically by VPA. In case of an overload of a replica, the VPA should provide and apply enhanced CPU and memory resources. Additionally, Gardener uses the cluster-autoscaler to upscale/downscale Nodes dynamically. Hence, it is not possible to ensure a balanced of allocatable CPU accross the zones.
```